py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -161,12 +163,26 @@ def compute_expected_flux(self, forests):
         forests = pool.map(self.read_true_continuum, forests)
         pool.close()
 
-        self.compute_mean_cont(forests)
+        for iteration in range(self.num_iterations):
```

Comment:
We wanted to take the output delta from the true continuum in order to obtain a var_lss which includes the smoothing from the spectrograph. This is done in the method compute_var_lss: 
https://github.com/igmhub/picca/blob/d851aa286a4619582be7d02216e0a299c437520c/py/picca/delta_extraction/expected_fluxes/true_continuum.py#L427

The idea of having iterations is to re-run true_continuum with the new var_lss. We could maybe just make two iterations and that's it:
- First iteration to compute the real smoothed var_lss value.
- Second iteration to perform the actual analysis.
Because the output var_lss in the second iteration is really close to the output in the first one.

Reply:
Ok, so if the only reason of doing iterations here is getting the varlss after cont fitting and then a mild modification of the mean_cont based on the actual weighting of the pixels, I'd suggest doing it explicitely as @iprafols suggested. And maybe even leave off the second mean_cont computation in case of e.g. constant_weights

Reply:
I have implemented the changes. As expected the mean continuum was slightly changed (percent level changes), so I also updated the test suite